moveit_py/src/moveit/moveit_ros/trajectory_execution_manager/trajectory_execution_manager.cpp

```diff
@@ -0,0 +1,224 @@
+/*********************************************************************
+ * Software License Agreement (BSD License)
+ *
+ *  Copyright (c) 2023, Matthijs van der Burgh
+ *  All rights reserved.
+ *
+ *  Redistribution and use in source and binary forms, with or without
+ *  modification, are permitted provided that the following conditions
+ *  are met:
+ *
+ *   * Redistributions of source code must retain the above copyright
+ *     notice, this list of conditions and the following disclaimer.
+ *   * Redistributions in binary form must reproduce the above
+ *     copyright notice, this list of conditions and the following
+ *     disclaimer in the documentation and/or other materials provided
+ *     with the distribution.
+ *   * Neither the name of the copyright holder nor the names of its
+ *     contributors may be used to endorse or promote products derived
+ *     from this software without specific prior written permission.
+ *
+ *  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ *  "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ *  LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+ *  FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+ *  COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+ *  INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+ *  BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ *  LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ *  CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ *  LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ *  ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ *  POSSIBILITY OF SUCH DAMAGE.
+ *********************************************************************/
+
+/* Author: Matthijs van der Burgh */
+
+#include "trajectory_execution_manager.h"
+
+namespace moveit_py
+{
+namespace bind_trajectory_execution_manager
+{
+static const rclcpp::Logger LOGGER = rclcpp::get_logger("moveit_py.bind_trajectory_execution_manager");
+
+void initTrajectoryExecutionManager(py::module& m)
+{
+  py::class_<trajectory_execution_manager::TrajectoryExecutionManager,
+             trajectory_execution_manager::TrajectoryExecutionManagerPtr>(m, "TrajectoryExecutionManager", R"(
+      Manages the trajectory execution.
+      )")
+
+      .def("isManagingControllers", &trajectory_execution_manager::TrajectoryExecutionManager::isManagingControllers,
```

Comment:
```suggestion
      .def("is_managing_controllers", &trajectory_execution_manager::TrajectoryExecutionManager::isManagingControllers,
```
Shouldn't the python functions be defined in `snake_case` (`camelCase` is used for cpp functions)? I think it is done like this everywhere else, so we should be consistent about it. Can you update the other function signatures as well?

Reply:
We need to update many of the existing cpp methods as there are discrepancies elsewhere in the library which may have lead to @MatthijsBurgh using different conventions. 

Not sure if there is an easy way to add this to auto formatter (clang-format).

Reply:
Added issue: https://github.com/ros-planning/moveit2/issues/2499

Reply:
I think that got covered by https://github.com/ros-planning/moveit2/pull/2177 or do you mean something else @peterdavidfagan. That PR also added clang-tidy checks to warn about cpp function names that don't follow the conventions. But don't think there are checks to catch  `.def("...", ..)` errors

Reply:
[49052c1](https://github.com/ros-planning/moveit2/pull/2177/commits/49052c11eb0ac20e0b7df2c3826721dfadb568ef) ❤️. 

My fork was outdated only seeing this updates now. Thanks so much.

As for `.def` I would personally need to become more familiar with `clang-tidy`, maybe there is a way to get a rule which also flags `.def`. For now I am happy to be more vigilant about this when pushing code or reviewing.

Reply:
Awesome, thanks for opening the issue though!

Reply:
So the python functions will be snake_case?